cmake/FindGLPK.cmake

```diff
@@ -16,28 +16,116 @@
 # GLPK_LIBRARIES - Libraries needed to use GLPK
 ##
 
-
 find_path(GLPK_INCLUDE_DIR NAMES glpk.h)
 find_library(GLPK_LIBRARIES NAMES glpk)
 
-# Check if we really have GLPK-cut-log.
-if(GLPK_INCLUDE_DIR)
-  include(CheckSymbolExists)
-  set(CMAKE_REQUIRED_INCLUDES ${GLPK_INCLUDE_DIR})
-  set(CMAKE_REQUIRED_LIBRARIES ${GLPK_LIBRARIES} m)
-  check_symbol_exists(glp_ios_get_cut "glpk.h" HAVE_GLPK_CUT_LOG)
-  if(NOT HAVE_GLPK_CUT_LOG)
-    message(FATAL_ERROR "Could not link against GLPK-cut-log. "
-                        "Did you forget to install GLPK-cut-log?")
+set(GLPK_FOUND_SYSTEM FALSE)
+if(GLPK_INCLUDE_DIR AND GLPK_LIBRARIES)
+  include(checksymbolexists)
+  set(cmake_required_includes ${glpk_include_dir})
+  set(cmake_required_libraries ${glpk_libraries} m)
+  check_symbol_exists(glp_ios_get_cut "glpk.h" have_glpk_cut_log)
+  if(not have_glpk_cut_log)
+    message(fatal_error "could not link against glpk-cut-log. "
+                        "did you forget to install glpk-cut-log?")
+  endif()
+  set(GLPK_FOUND_SYSTEM TRUE)
+endif()
+
+if(NOT GLPK_FOUND_SYSTEM)
+  check_ep_downloaded("GLPK-EP")
+  if(NOT GLPK-EP_DOWNLOADED)
+    check_auto_download("GLPK" "--no-glpk")
+  endif()
+
+  include(ExternalProject)
+  include(GNUInstallDirs)
+
+  set(GLPK_VERSION "5.0")
+
+  if("${CMAKE_GENERATOR}" STREQUAL "Unix Makefiles")
+    # use $(MAKE) instead of "make" to allow for parallel builds
+    set(make_cmd "$(MAKE)")
+  else()
+    # $(MAKE) does not work with ninja
+    set(make_cmd "make")
+  endif()
+
+  find_package(Patch)
+  if(NOT Patch_FOUND)
+    message(SEND_ERROR "Can not build GLPK, missing binary for patch")
```

Comment:
I think all of these should be `FATAL_ERROR`, but it matches what's done here:

- https://github.com/cvc5/cvc5/blob/6a3ffc465718fb847f3c37fca7e6598171eb23ac/cmake/FindCLN.cmake#L55-L58